### PR TITLE
Set existing torrent webseeds after download

### PR DESCRIPTION
**Pull request by mh0lt**
Fix a timing hole where torrents that get created before webseeds have been downloaded don't get webseeds set.

### CODE REVIEW

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]
We already have place in code - with meaning "do things after webseeds discovered": 
- it's in `mainLoop`: 
```
go func() {
			defer d.wg.Done()
			// webseeds.Discover may create new .torrent files on disk
			d.webseeds.Discover(d.ctx, d.cfg.WebSeedFiles, d.cfg.Dirs.Snap)
			// apply webseeds to existing torrents
			if err := d.addTorrentFilesFromDisk(true); err != nil && !errors.Is(err, context.Canceled) {
				d.logger.Warn("[snapshots] addTorrentFilesFromDisk", "err", err)
			}
		}()
```
so, i think better add explicit handling for this case here. 
  
also maybe here need replace `.addTorrentFilesFromDisk` by `for client.Torrents()` loop. (because we already added files from disk in constructor).

**Comment by AskAlexSharov** [comment]
let's add 1 unit-test for this case

**Comment by mh0lt** [comment]
> d.webseeds.Discover

I think we need both cases to make sure webseeds get included everywhere as I think there are - I'd prefer to leave this in at the moment as the execution path will be the same as for the testing I did yesterday.

I can add a  unit tests - but the as there don't seem to be any webseed related tests at the moment - I'll do that in a separate PR as I'll need to code a webseed mock.

File: erigon-lib/downloader/downloader.go [hunk 1]
```diff
@@ -821,6 +821,15 @@ func (d *Downloader) mainLoop(silent bool) error {
 			if err := d.addTorrentFilesFromDisk(true); err != nil && !errors.Is(err, context.Canceled) {
 				d.logger.Warn("[snapshots] addTorrentFilesFromDisk", "err", err)
 			}
+
+			d.lock.Lock()
```
Comment (AskAlexSharov) [inline erigon-lib/downloader/downloader.go:825]: move it to some function in future PR

**Review by AskAlexSharov — APPROVED** [review]

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**